File: Golang/3099-harshad-number/harshad-number.go
func sumOfTheDigitsOfHarshadNumber(x int) int {
    sum, temp := 0, x
    for temp > 0 {
        sum += temp % 10
        temp /= 10
    }
    if x % sum == 0 {
        return sum
    }
    return -1
}

File: Python/3371-harshad-number/harshad-number.py
class Solution:
    def sumOfTheDigitsOfHarshadNumber(self, x: int) -> int:
        digit_sum = 0
        temp = x
        while temp > 0:
            digit_sum += (temp % 10)
            temp = temp // 10
        if x % digit_sum ==  0:
            return digit_sum
        return -1